opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/aeseh/AdditiveEpsilonMapping.java

```diff
@@ -0,0 +1,69 @@
+package org.opt4j.optimizers.ea.aeseh;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.opt4j.core.Individual;
+import org.opt4j.core.Objective;
+import org.opt4j.core.Objective.Sign;
+import org.opt4j.core.Objectives;
+
+/**
+ * Implements the evenly spaced adaptive ε function.
+ * 
+ * @author Fedor Smirnov
+ *
+ */
+public class AdditiveEpsilonMapping implements EpsilonMapping {
+
+	@Override
+	public Objectives mapObjectives(Objectives original, double epsilon, Map<Objective, Double> objectiveAmplitudes) {
+		Objectives result = new Objectives();
+		Iterator<Objective> iterator = original.getKeys().iterator();
+		double[] values = original.array();
+		for (int i = 0; i < original.size(); i++) {
+			Objective obj = iterator.next();
+			double value = values[i] * (obj.getSign().equals(Sign.MIN) ? 1 : -1);
+			if (objectiveAmplitudes.containsKey(obj)) {
+				// the ε mapping is only applied if the objective is feasible for at least one individual
+				value += epsilon * objectiveAmplitudes.get(obj) * (obj.getSign().equals(Sign.MAX) ? 1 : -1);
+			}
+			result.add(obj, value);
+		}
+		return result;
+	}
+
+	@Override
+	public Map<Objective, Double> findObjectiveAmplitudes(Set<Individual> individuals) {
```

Comment:
Javadoc comment

Reply:
At this point I think that the javadocs generated for the interface of the class do provide an exact description of what the method does so that no additional comments are necessary.

Reply:
Correct, sorry.